File: KuantumKaosYonetimi/CSharp/KuantumKaosYonetimi/KuantumCokusuException.cs
namespace KuantumKaosYonetimi
{
    // D. Özel Hata Yönetimi (Custom Exception)
    // Standart Exception sınıfından türetilir.
    public class KuantumCokusuException : Exception
    {
        public KuantumCokusuException(string nesneId)
            : base($"Kuantum Çöküşü! Patlayan Nesne ID: {nesneId}")
        {
        }
    }
}

File: KuantumKaosYonetimi/Java/KuantumKaosYonetimi/KuantumCokusuException.java
// KuantumCokusuException.java
package KuantumKaosYonetimi;

// RuntimeException'dan türetilir (try-catch gereksinimi için)
public class KuantumCokusuException extends RuntimeException {
    
    // Yapıcı metot
    public KuantumCokusuException(String nesneId) {
        super("Kuantum Çöküşü! Patlayan Nesne ID: " + nesneId);
    }
}